sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/PartitionKey.java

```diff
@@ -25,7 +25,13 @@
      */
     @SuppressWarnings("serial")
     public PartitionKey(final Object key) {
-        this.internalPartitionKey = PartitionKeyInternal.fromObjectArray(new Object[] {key}, true);
+        // Now that we are making partitionkey as one of the parameters for creating object, just making sure nothing
+        // wrong happens here when user does something like new PartitionKey(PartitionKey.None);
+        if (!(key instanceof PartitionKey)) {
```

Comment:
if the intention is to only capture PartitionKey.None then we should probably do PartitionKey.None == key instead.

Are there any other case other than PartitionKey.None which would get covered here?

Reply:
Removed this check altogether. Will add if we see many people using this wrongly.